From: Loic Dachary <loic-201408@dachary.org>
Date: Thu, 9 Oct 2014 18:52:17 +0200
Subject: [PATCH] ceph-disk: run partprobe after zap

Not running partprobe after zapping a device can lead to the following:

* ceph-disk prepare /dev/loop2
* links are created in /dev/disk/by-partuuid
* ceph-disk zap /dev/loop2
* links are not removed from /dev/disk/by-partuuid
* ceph-disk prepare /dev/loop2
* some links are not created in /dev/disk/by-partuuid

This is assuming there is a bug in the way udev events are handled by
the operating system.

http://tracker.ceph.com/issues/9665 Fixes: #9665

Signed-off-by: Loic Dachary <loic-201408@dachary.org>
(cherry picked from commit fed3b06c47a5ef22cb3514c7647544120086d1e7)
---
 src/ceph-disk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ceph-disk b/src/ceph-disk
index 944caf8..fb4400b 100755
--- a/src/ceph-disk
+++ b/src/ceph-disk
@@ -1066,6 +1066,9 @@ def zap(dev):
                 dev,
             ],
         )
+
+        update_partition('-d', dev, 'zapped')
+
     except subprocess.CalledProcessError as e:
         raise Error(e)
 
